flink-table-store-core/src/main/java/org/apache/flink/table/store/file/writer/RecordWriter.java

```diff
@@ -32,9 +33,18 @@
  */
 public interface RecordWriter<T> {
 
+    /** Open the record write. */
+    void open(MemorySegmentPool memoryPool);
```

Comment:
No need to change the base interface. These methods are only useful in `KeyValueRecordWriter`.

Reply:
The current AppendOnlyWriter also has memory problems. When there are too many partitions, the ORC Writer's VectorizedRowBatch will also take up a lot of memory, leading to OOM.
I'm considering whether I can also bring to AppendOnly in the future the ability to.
- ORC writer's memory can be pooled
- Or put records into MemorySegment first, then write the file

So I introduce this to `RecordWriter` interface.